laws/src/main/scala/cats/laws/discipline/arbitrary.scala

```diff
@@ -130,7 +130,20 @@ object arbitrary extends ArbitraryInstances0 with ScalaVersionSpecific.Arbitrary
       a <- A.arbitrary
     } yield NonEmptyMap((k, a), fa))
 
-  implicit def cogenNonEmptyMap[K: Order: Cogen, A: Order: Cogen]: Cogen[NonEmptyMap[K, A]] =
+  @deprecated("Preserved for bincompat", "2.9.0")
+  def cogenNonEmptyMap[K, A](kOrder: Order[K],
+                             kCogen: Cogen[K],
+                             aOrder: Order[A],
+                             aCogen: Cogen[A]
+  ): Cogen[NonEmptyMap[K, A]] = {
+    implicit val orderingK: Order[K] = kOrder
+    implicit val cogenK: Cogen[K] = kCogen
+    implicit val cogenA: Cogen[A] = aCogen
+
+    cogenNonEmptyMap[K, A]
+  }
+
+  implicit def cogenNonEmptyMap[K: Order: Cogen, A: Cogen]: Cogen[NonEmptyMap[K, A]] =
```

Comment:
I'm confused why we even need the `K: Order` here. A `SortedMap` knows the `Ordering[K]`.

Doesn't this also work only with `K: Cogen, A: Cogen`?

Reply:
I was confused about this as well :) Scalacheck implements the `Cogen[Map]` via `Cogen[Vector]`.

https://github.com/typelevel/scalacheck/blob/7bb1a47af2313f7e7935d34cce255f7b7c0b1b31/core/shared/src/main/scala/org/scalacheck/Cogen.scala#L129-L130

Reply:
Good catch! As a side note: in fact, there are quite a lot of places in the Cats codebase where a constraint is demanded but not actually used. So this is why I think it would be great to have "unused" warnings enforced eventually.